chainercv/datasets/sbd/sbd_utils.py

```diff
@@ -0,0 +1,70 @@
+import os
+import six
+
+from chainer.dataset import download
+
+from chainercv import utils
+
+root = 'pfnet/chainercv/sbd'
+url = 'http://www.eecs.berkeley.edu/Research/Projects/CS/vision/grouping/semantic_contours/benchmark.tgz'  # NOQA
+train_voc2012_url = 'http://home.bharathh.info/pubs/codes/SBD/train_noval.txt'
+
+
+def get_sbd():
+    data_root = download.get_dataset_directory(root)
+    base_path = os.path.join(data_root, 'benchmark_RELEASE/dataset')
+    train_voc2012_file = os.path.join(base_path, 'train_voc2012.txt')
+    if not os.path.exists(train_voc2012_file):
+        six.moves.urllib.request.urlretrieve(
+            train_voc2012_url, train_voc2012_file)
+
+    if os.path.exists(os.path.join(base_path, 'train.txt')):
+        # skip downloading
+        return base_path
+
+    download_file_path = utils.cached_download(url)
+    ext = os.path.splitext(url)[1]
+    utils.extractall(download_file_path, data_root, ext)
+    generate_voc2012_txt(base_path)
+
+    return base_path
+
+
+def generate_voc2012_txt(base_path):
+    with open(os.path.join(base_path, 'train.txt'), 'r') as f:
+        train_ids = f.read().split('\n')[:-1]
+    with open(os.path.join(base_path, 'val.txt'), 'r') as f:
+        val_ids = f.read().split('\n')[:-1]
+    with open(os.path.join(base_path, 'train_voc2012.txt'), 'r') as f:
+        train_voc2012_ids = f.read().split('\n')[:-1]
+    all_ids = list(set(train_ids + val_ids))
+    val_voc2012_ids = [i for i in all_ids if i not in train_voc2012_ids]
+
+    with open(os.path.join(base_path, 'val_voc2012.txt'), 'w') as f:
+        f.write('\n'.join(sorted(val_voc2012_ids)))
+    with open(os.path.join(base_path, 'trainval_voc2012.txt'), 'w') as f:
+        f.write('\n'.join(sorted(all_ids)))
+
+
+sbd_instance_segmentation_label_names = (
+    'background',
```

Comment:
The data representation looks okay except for the range of `label`, which I want it to be `[0, n_fg_class - 1]`. `n_fg_class` is the number of classes excluding the background. This means that the label would never be "background". The same convention is adopted for detection.

I think that consistency really matters between detection and instance segmentation because two of them have very similar interface.

Reply:
i add background as semantic segmentation because we normally have background label in instance segmentation.
The task is pixelwise segmentation and it means that label includes baxkground for all pixels.

Reply:
now Mask-RCNN and FCIS have two stage structure with RPN, but this task predicts pixelwise probabilities, not detecting the object.

Reply:
These methods are OK with no-background label names, so I updated to remove background label.
However, we need to think better way to merge semantic segmentation, instance segmentation and object detection.